dotnet/src/SemanticKernel.Abstractions/Orchestration/FunctionResult.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.SemanticKernel.Orchestration;
+
+/// <summary>
+/// Function result after execution.
+/// </summary>
+public sealed class FunctionResult
+{
+    /// <summary>
+    /// Additional data from model.
+    /// </summary>
+    public IReadOnlyCollection<ModelResult> ModelResults { get; internal set; } = Array.Empty<ModelResult>();
```

Comment:
Do we expect NativeFunctions to populate this?

Reply:
No, it is specifically for AI model additional data during semantic function execution, like token usage for example. However, since `ModelResult` also contains `object` property, we may consider making it more generic (e.g. `Dictionary<string, object> ExtensionData`, which will also allow native functions to put some additional metadata during function execution if required.

I would probably make this change as part of separate PR, to keep this one as lightweight as possible.

Reply:
@dmytrostruk, suggest to remove this property in thie PR and add it with the ExtensionData approach later to avoid introducing something that we will break right after.

Reply:
I decided to add this in scope of this PR, the implementation will be ready soon :)

Reply:
I would propose naming the property 'Metadata' if we decide to go with it so that it's clear that it contains data describing the result, while the result itself is represented by the 'value' property.

Reply:
Thanks for all the feedback! I replaced `ModelResults` with `Metadata`.